tools/xdp/s2n-quic-xdp/src/ring/cursor.rs

```diff
@@ -138,19 +143,19 @@ impl<T: Copy + fmt::Debug> Cursor<T> {
             return free;
         }
 
-        let new_value = self.consumer().load(Ordering::Acquire);
+        let mut new_value = self.consumer().load(Ordering::Acquire);
+
+        // Our cached copy has the size added so we also need to add the size here when comparing
+        //
+        // See `Self::init_producer` for more details
+        new_value += self.size;
```

Comment:
This was a fun bug to hunt down... The optimization that I put in to compare the `new_value` against the cached value backfired and caused a deadlock when the consumer drained the queue entirely.